src/map_extras.cpp

```diff
@@ -118,17 +128,17 @@ void mx_helicopter( map &m, const tripoint &abs_sub )
     int x_min = abs( bbox.p1.x ) + 0;
     int y_min = abs( bbox.p1.y ) + 0;
 
-    int x_max = ( SEEX * 2 ) - ( bbox.p2.x + 1 );
-    int y_max = ( SEEX * 2 ) - ( bbox.p2.y + 1 );
+    int x_max = ( map_tiles_width ) - ( bbox.p2.x + 1 );
+    int y_max = ( map_tiles_width ) - ( bbox.p2.y + 1 );
 
     int x1 = clamp( cx + x_offset, x_min,
                     x_max ); // Clamp x1 & y1 such that no parts of the vehicle extend
     int y1 = clamp( cy + y_offset, y_min, y_max ); // over the border of the submap.
 
-    vehicle *wreckage = m.add_vehicle( crashed_hull, tripoint( x1, y1, abs_sub.z ), dir1, rng( 1, 33 ),
+    vehicle *wreckage = tmp_m.add_vehicle( crashed_hull, tripoint( x1, y1, abs_sub.z ), dir1, rng( 1, 33 ),
                                        1 );
 
-    if( wreckage != nullptr ) {
+    if( wreckage != nullptr && true ) {
```

Comment:
Hopefully you're going to revert that change, because it will lead to UB as `wreckage` is dereferenced below.